modules/options.nix

```diff
@@ -61,6 +61,37 @@ let
 
   updateOptions = _: {
     options = {
+      restartDelay = mkOption {
```

Comment:
I was not sure where to put them, so I just went and added them to the update attrset, there is nospecifi reason whatsoever :)
I think that a separate attrset like the one you proposed it the best idea honestly!
I am not great a naming thing either, but I think `onServiceFailure` or `restartOnFailure` are good names. 

Since this will be a separate attrset, we could also have an `enable` option, which when disabled, rolls everything back to the behaviour we have in the last release, what do you think?

Reply:
> I am not great a naming thing either, but I think onServiceFailure or restartOnFailure are good names.

Alright! Then maybe I'd lean towards  `onServiceFailure`, to match existing naming conventions. Good naming or not, at least we are consistent :).

> Since this will be a separate attrset, we could also have an enable option, which when disabled, rolls everything back to the behaviour we have in the last release, what do you think?

+1 for an `enable` toggle.

Reply:
I just imeplemented these changes and I went with `restartOnFailure` as a name.
Let me know what do you think once you get a chance to review the commit!

And happy holidays!